src/common/src/system_param.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+
+use paste::paste;
+use risingwave_pb::meta::SystemParams;
+
+use crate::error::{ErrorCode, RwError};
+
+type Result<T> = core::result::Result<T, RwError>;
+
+// Includes deprecated params. Used to define key constants.
+macro_rules! for_all_params {
```

Comment:
Will it be simpler to make a list of deprecated params? So that there will be 2 macros:

- `for_all_undeprecated_params` (or a shorter name?)
- `for_all_deprecated_params`

Then `for_all_params` can be done by calling both of them.